src/workflows-app/components/FilterableWorkflowTable.ts

```diff
@@ -43,37 +96,33 @@ const FilterableWorkflowTable = ({
   namespace,
   submissionId,
   workspaceName,
+  workspaceId,
+  setLogsModal,
+  setTaskDataModal,
 }: FilterableWorkflowTableProps) => {
   const [itemsPerPage, setItemsPerPage] = useState(50);
   const [filterOption, setFilterOption] = useState<string>();
   const [pageNumber, setPageNumber] = useState(1);
   const [sort, setSort] = useState({ field: 'duration', direction: 'desc' });
   const [viewErrorsId, setViewErrorsId] = useState<number>();
+  const [currentWorkflow, setCurrentWorkflow] = useState<WorkflowMetadata>();
 
   const errorStates = ['SYSTEM_ERROR', 'EXECUTOR_ERROR'];
+  const signal = useCancellation();
 
-  const getFilteredRuns = (filterOption: string, runsData: Run[]): Run[] => {
-    const filteredRuns: Run[] = [];
-
-    for (const run of runsData) {
-      switch (filterOption) {
-        case 'Error':
-          if (errorStates.includes(run.state)) {
-            filteredRuns.push(run);
-          }
-          break;
-        case 'Succeeded':
-          if (run.state === 'COMPLETE') {
-            filteredRuns.push(run);
-          }
-          break;
-        default:
-          filteredRuns.push(run);
-      }
-    }
+  const showLogModal = useCallback(
+    (modalTitle: string, logsArray: LogInfo[]) => {
+      setLogsModal({ modalTitle, logsArray });
+    },
+    [setLogsModal]
+  );
```

Comment:
Nit: Since this only slightly reshapes arguments, is the extra function here worth it vs directly using `setLogsModal`?